gha-webhook-listener.py

```diff
@@ -74,21 +72,40 @@ def req_headers():
     }
 
 
-def validate_signature(data, signature, token):
-    github_secret = bytes(token, 'utf-8')
+def validate_signature(data: bytes, signature: str, token: str) -> bool:
+    """
+    Checks that the request comes from a trusted source.
+
+    Args:
+        data: data which the signature ought to be applied to
+        signature: signature that was sent in the request headers
+        token: the secret we know and expect Github to use to sign data
+
+    Returns:
+        Whether the signature matches our expectations
+    """
+    github_secret = token.encode('utf-8')
     expected_signature = hmac.new(key=github_secret, msg=data, digestmod=hashlib.sha256).hexdigest()
-    signature = signature.split('sha256=')[-1].strip()
-    return hmac.compare_digest(signature, expected_signature)
+    if not signature.startswith('sha256='):
+        return False
+    else:
+        signature = signature.split('sha256=')[-1].strip()
+        return hmac.compare_digest(signature, expected_signature)
```

Comment:
```suggestion
    signature = signature[7:].strip()
    return hmac.compare_digest(signature, expected_signature)
```

Reply:
Funny, I initially wrote like this. I don’t know why I changed it. :smile: